bokehjs/src/lib/models/annotations/legend.ts

```diff
@@ -256,15 +263,21 @@ export class LegendView extends AnnotationView {
       })()
 
       for (const label of labels) {
+        // glyph and label are drawn side-by-side in a single box, such that:
+        // (x1, y1): top-left of glyph
+        // (x2, y2): bottom left of text label
+        // TODO: in order to get vertical alignment working right, glyph and label will have to be drawn in separate
+        // side-by-side boxes, with separate yoffsets for glyphs (so that glyphs smaller than max_glyph_size can be
+        // scooted down by size difference / 2) and label (standard yoffset as calculated below). --cfh
         const x1 = bbox.x + xoffset
         const y1 = bbox.y + yoffset + this.title_height
-        const x2 = x1 + glyph_width
+        const x2 = x1 + max_glyph_size
         const y2 = y1 + glyph_height
```

Comment:
This would also need to be `max_glyph_size` for the `y2` I think? I think that explains the vertical offsetting in the screenshots. But also see discussion about width/height values